server/lib/oauth2/oauth2.executeOauth2HTTPQuery.js

```diff
@@ -0,0 +1,140 @@
+  const authorizeHost = await variable.getValue(OAUTH2.VARIABLE.AUTHORIZE_HOST, serviceId);
+  const authorizePath = await variable.getValue(OAUTH2.VARIABLE.AUTHORIZE_PATH, serviceId);
+  const additionalAccessTokenRequestActionParam = await variable.getValue(
+    OAUTH2.VARIABLE.ADDITIONAL_ACCESS_TOKEN_REQUEST_ACTION_PARAM,
+    serviceId,
+  );
+
+  // Init credentials based on integration name
+  const credentials = {
+    client: {
+      id: clientId,
+      secret,
+    },
+    auth: {
+      tokenHost,
+      tokenPath,
+      authorizeHost,
+      authorizePath,
+    },
+    http: {
+      json: true,
+    },
+  };
+
+  try {
+    const client = new ClientCredentials(credentials);
+    let authResult = await client.createToken(accessToken);
+
+    if (authResult.expired()) {
+      logger.trace('Refresh token is requiered');
+      // Refresh token
+      try {
+        const refreshParams = {
+          client_id: clientId,
+          client_secret: secret,
+          grant_type: 'refresh_token',
+          refresh_token: authResult.token.refresh_token,
+        };
+        if (additionalAccessTokenRequestActionParam) {
+          refreshParams.action = additionalAccessTokenRequestActionParam;
+        }
+
+        authResult = await authResult.refresh(refreshParams);
+
+        // Save new  accessToken
+        if (!authResult.token.status || authResult.token.status === 0) {
+          let jsonResult;
+          if (authResult.token.body) {
+            jsonResult = authResult.token.body;
+            authResult = await client.createToken(jsonResult);
+          } else {
+            jsonResult = authResult.token;
+          }
+          await variable.setValue(OAUTH2.VARIABLE.ACCESS_TOKEN, JSON.stringify(jsonResult), serviceId, userId);
+        }
+      } catch (error) {
+        logger.error('Error refreshing access token: ', error);
+      }
+    }
+
+    logger.trace('authResult: ', authResult);
+    return authResult;
+  } catch (error) {
+    logger.error(error);
+    return null;
+  }
+}
+
+/**
+ * @description Execute a query to Oauth2 API.
+ * @param {string} serviceId - Gladys service id call method.
+ * @param {string} userId - Gladys userId of current session.
+ * @param {string} queryType - Method of call ('get' or 'post').
+ * @param {string} queryUrl - Url to send query.
+ * @param {string} queryParams - Array of query params.
+ * @returns {Promise} Result of query .
+ * @example
+ * oauth2.executeOauth2HTTPQuery('7fdsf4s68r4gfr68f4r63csd7f6f4c3r85', '78v4f3df83g74v1fsd8375f63gvrf5c', 'withings'
+ *  'Bearer', 'get', 'http://localhost/apiname',[...]);
+ */
+async function executeOauth2HTTPQuery(serviceId, userId, queryType, queryUrl, queryParams) {
+  // Refresh token access if needed
+  const accesToken = await refreshTokenAccess(this.variable, serviceId, userId);
+  const headerConfig = {
+    headers: {
+      Authorization: `${accesToken.token.token_type} ${accesToken.token.access_token}`,
```

Comment:
Can you do a search on the whole codebase, and change `accesToken` (typo) with `accessToken` ?

Reply:
i fix it